### PR TITLE
feat(kafka): add sasl auth support

### PR DESCRIPTION
**Pull request by andrewthauer**
## Hey, I just made a Pull Request!

Adds SASL & SSL/TLS boolean config support for `kafka-backend` plugin.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kafka-backend/src/config/ClusterReader.ts [hunk 1]
```diff
@@ -15,23 +15,21 @@
...
-import { ClusterDetails } from '../types/types';
+import { ClusterDetails, SslConfig, SaslConfig } from '../types/types';
 
 export function getClusterDetails(config: Config[]): ClusterDetails[] {
   return config.map(clusterConfig => {
     const clusterDetails = {
       name: clusterConfig.getString('name'),
       brokers: clusterConfig.getStringArray('brokers'),
     };
-    const sslConfig = clusterConfig.getOptional('kafka.ssl');
```
Comment (andrewthauer) [inline plugins/kafka-backend/src/config/ClusterReader.ts:27]: This nested cluster key `kafka.ssl` seems to have been incorrect. It doesn't match what is defined in the `config.d.ts` file. Not sure if this was a mistake or not. @nirga - Can you confirm this?

File: plugins/kafka-backend/config.d.ts [hunk 2]
```diff
@@ -29,11 +29,21 @@ export interface Config {
...
+            cert: string;
+          }
+        | boolean;
+      /**
+       * Optional SASL connection parameters.
+       */
+      sasl?: {
+        mechanism: 'plain' | 'scram-sha-256' | 'scram-sha-512';
+        username: string;
+        password: string;
```
Comment (benjdlambert) [inline plugins/kafka-backend/config.d.ts:46]: `/** @visibility secret */`?

**Review by benjdlambert — APPROVED** [review]

File: plugins/kafka-backend/README.md [hunk 3]
```diff
@@ -28,3 +28,21 @@ kafka:
...
       - localhost:9092
 ```
+
+### `sasl` (optional)
+
+Configure SASL authentication for the Kafka client.
+
+```yaml
+kafka:
+  clientId: backstage
```
Comment (nirga) [inline plugins/kafka-backend/README.md:38]:
```suggestion
  - clientId: backstage
```

File: plugins/kafka-backend/README.md [hunk 4]
```diff
@@ -28,3 +28,21 @@ kafka:
...
+  clusters:
+    name: prod
```
Comment (nirga) [inline plugins/kafka-backend/README.md:40]:
```suggestion
    - name: prod
```

**Pull request merged**